src/util/stft.rs

```diff
@@ -452,7 +488,7 @@ impl<const NUM_SIDECHAIN_INPUTS: usize> StftHelper<NUM_SIDECHAIN_INPUTS> {
                     // longer than the block size, then this will cause everything else to be
                     // shifted to the left so it can be added in the iteration after this.
                     if self.padding > 0 {
-                        let padding_to_copy = cmp::min(self.padding, block_size);
+                        let padding_to_copy = cmp::max(self.padding, block_size);
```

Comment:
This is incorrect. If the block size is 2048, and the amount of padding is 512 samples, then the 512 sample tail from the last iteration (stored in `padding_buffer`) should be copied to the output (the start of `self.scratch_buffer`). Your code c hanges it so that 2048 samples of padding are copied from `padding_buffer`, which results in a panic since `padding_buffer` is only 512 samples long.